src/iwant_bot/start.py

```diff
@@ -13,104 +14,134 @@
+        print('WARNING: MultiDict contains duplicate keys, last occurrence was used.')
+        print(multidict)
+    return {key: multidict[key] for key in multidict}
 
 
-def body_to_dict(body):
-    """Try to convert MultiDictProxy to dictionary.
-    Not handled ValueError"""
-    keys = set(body.keys())
-    if len(keys) == len(body):
-        body_dict = {}
-        for key in keys:
-            body_dict[key] = body[key]
-    else:
-        raise ValueError('MultiDict contains same keys.')
-    return body_dict
+async def handle_slack_iwant(request):
+    body = multidict_to_dict(await request.post())
+    body['incoming_ts'] = time.time()
+    print(f'INFO: iwant request body:\n{body}.')
 
+    try:
+        verify_request_token(body)
+    except (KeyError, TokenError) as err:
+        print(f'INFO: Invalid token: {err}')
+        return web.json_response({'text': 'Unverified message.'})
 
-def trigger_reaction(body, trigger):
-    """Commands: /iwant,
-    Trigger_words: None"""
-    message = format_response(
-        f"{body['user_name']} used {body[trigger]} with {body['text']}.")
-    return web.json_response(body=message)
+    if 'command' in body:
+        print(f"INFO: iwant handler handles command '{body['command']}'")
+    else:
+        print("WARNING: Request does not specify 'command'.")
+        return web.json_response({'text': 'Tried to handle command, but none found.'})
+
+    iwant_object = IwantRequest(body, _iwant_activities, _iwant_behest, _slack_user_pattern,
+                                _other_words, _default_duration, _max_duration)
+    print(f'INFO: iwant parsed request:\n{iwant_object.data}')
+
+    # Check and process the behests
+    res = handle_slack_iwant_behest(iwant_object)
+    if res is not None:
+        return web.json_response(res)
+
+    # If no behest, then resolve the activities
+    return web.json_response(handle_slack_iwant_command(iwant_object))
+
+
+def handle_slack_iwant_behest(iwant_object) -> dict or None:
+    if len(iwant_object.data['behests']) > 1:
+        print("INFO: More than 1 behest was found.")
+        return {'text': f"You can use only one from `{'`, `'.join(iwant_object.possible_behests)}`"
+                        " at the same time."}
+    elif len(iwant_object.data['behests']) == 1:
+        print(f"INFO: iwant request found behest '{iwant_object.data['behests'][0]}'.")
+        if iwant_object.data['behests'] == ['list']:
+            return iwant_object.return_list_of_parameters()
+        elif iwant_object.data['behests'] == ['help']:
+            return iwant_object.create_help_message()
+        # other behests
+        else:
+            return {'text': f"{iwant_object.data['behests'][0]}"
+                            " is not implemented yet. Coming soon!"}
+    else:
+        return None
 
 
-async def handle_post(request):
-    body = body_to_dict(await request.post())
-    print(body)
+def handle_slack_iwant_command(iwant_object) -> dict:
```

Comment:
It is OK, but it is difficult to visually separate error handling from the single case when the input is actually OK.